packages/core/src/State.ts

```diff
@@ -245,6 +251,27 @@ export class State<
     return isInFinalState(this.configuration);
   }
 
+  /**
+   * The done data of the top-level finite state.
+   */
+  public get output(): any {
+    if (!this.done) {
+      return undefined;
+    }
+
+    const finalChildStateNode = this.configuration.find(
+      (stateNode) =>
+        stateNode.type === 'final' && stateNode.parent === this.machine?.root
```

Comment:
It feels that we should figure out changes to the API that will make `this.machine` required for `State`.

Reply:
#3551 was merged in, and `this.machine` is now required on `State` instances, so those changes were made ✅